### PR TITLE
Commerce: Item Updates!

### PR DESCRIPTION
**Pull request by zfox23**
This PR adds client support for Marketplace Item Updates. For example:
1. A Creator uploads `App v1.0`
2. User A purchases `App v1.0`
3. The creator later revises `App v1.0` to `App v2.0`
4. User A can now update their purchased Edition of `App v1.0` to `App v2.0`! Woohoo!

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8022/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8023/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8023/HighFidelity-Beta-PR12638-6647475f5cf1253687c77be2000af00d4a2e561c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8023/HighFidelity-Beta-PR12638-6647475f5cf1253687c77be2000af00d4a2e561c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8031/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8031/HighFidelity-Beta-PR12638-f3efdda6b020b916fcab90c25408e3977e3c2343.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8031/HighFidelity-Beta-PR12638-f3efdda6b020b916fcab90c25408e3977e3c2343.exe

**Comment by zfox23** [comment]
some bugfixes pushed:
1. minor logspam fix
2. fix updating costed items when you don't have enough money to buy it if you weren't updating
3. fix "number sold" display
4. fix "updatable" filter
5. potential fix for "endless spinning on checkout if wallet not auth'd when clicking on marketplace item"

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8037/app_12638_PR.apk

**Comment by zfox23** [comment]
there are two more things that are less bugs and more feature requests:
1. make use of `since` parameter for `available_updates` endpoint - there was some miscommunication about when that red light would be dismissed. i have to re-think the behavior.
2. if you organically get to the marketplace item page of an update to an item that you own, and then you click "free" or "buy", the Checkout page should display "buy it again" (if that's possible), AND "you own the previous version of this item, click here to view this item in my purchases and select the version you'd like to update"

(1) is hopefully relatively easy
(2) doesn't have a mockup or wireframe, but was indeed in the spec...

>However, if you do buy the new item in the normal way, where the confirmation screen checks to see if you already have it, it will also check to see if you you own the original. If either is true (you already own an update, or you already own an original), you are told this in text. (The text for already owning an update was lost in the Expanded Certified Content project, and will be brought back.) The single call to action for either is to show the matching items in your purchases, so the user can take specific action on those

i missed implementing (2) :(

for now, until we figure out what we actually want for (1), i'm going to make it so that clicking on a LIT "my purchases" link _doesn't_ cause the "updatable" filter to be applied to My Purchases

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8040/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8040/HighFidelity-Beta-PR12638-2d57c444b6d7dfa1e2b35ac948cb1a85ddef806e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8040/HighFidelity-Beta-PR12638-2d57c444b6d7dfa1e2b35ac948cb1a85ddef806e.dmg

**Comment by zfox23** [comment]
## Test Plan

## Notes
- _Most of_ this test is only required to be performed in Desktop mode; it is not necessary to run this entire test plan in both HMD and Desktop modes.
    - Certain steps during the test will be marked as needing to be performed in both HMD and Desktop modes.
    - If, at the tester's discretion, a step during the test appears as though it might behave differently in HMD, please let me know and perform the step in HMD mode.

## Prerequisites
- You have a High Fidelity account on the Staging backend environment
- Your environment variables are set up to point Interface/Sandbox at the Staging backend environment
    - `SETX HIFI_METAVERSE_URL https://staging.highfidelity.com`
- You have a valid Wallet that you have authenticated
    - Your Wallet does not need to contain any HFC
- You do not have any Purchases tied to your Wallet
- You are connected to your local Sandbox where you _DO_ have Rez Certified, Rez Temp Certified, and Change Content permissions

## Tests

### Test A: Wearable Purchase - Base Version
1. Open the Marketplace. Go to the item page for a wearable entity. One example of a wearable entity on the Staging Metaverse is "Monztur Hat v<X.X>", where <X.X> is some version number that'll change over time. Click "Purchase".
    i. Verify that you see the "Review Purchase" screen.
    ii. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "CONFIRM PURCHASE" button, and a borderless "CANCEL" button.
2. Click "CONFIRM PURCHASE".
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **wearable** [marketplace item] has been added..."
    iii. Verify that you see a "[hat icon] WEAR" button, and that it is enabled.
    iv. Verify that DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
3. Click "View this item in My Purchases"
    i. Verify that you are taken to the My Purchases screen, and that the item you just purchased appears at the top of the list.
    ii. Verify that you see a "[hat icon] WEAR" button on the card of the item you just purchased, and that the button is enabled.
    iii. Verify that you do not see a red dot to the left of "My Purchases" in the header, and that you do not see a red dot on the "MARKET" icon in your HUD.
4. Click the item name of the marketplace item you just purchased.
    i. Verify that you are returned to the marketplace item page for the item from (1).
5. Click "Purchase" on the marketplace item page.
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "VIEW THIS ITEM IN MY PURCHASES" button, a grey "BUY IT AGAIN" button, and a borderless "CANCEL" button.
6. Click "VIEW THIS ITEM IN MY PURCHASES"
    i. Verify that you are taken to the My Purchases screen, and that the filter bar contains the text corresponding to the title of the item you were about to buy.
7. Click the "x" icon inside the filter bar.
    i. Verify that the text inside the filter bar disappears.
8. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 0 items appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 0 items appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 0 items appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 0 items appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 0 items appear in the list after you have selected the "Updatable" category in the filter bar.
9. Rez the wearable entity.
    i. The entity appears on your avatar's body somewhere.
10. Inspect the wearable entity.
    i. The wearable's certificate is valid.

### Test B: Entity Purchase - Base Version
1. Open the Marketplace. Go to the item page for a regular entity. One example of a regular entity on the Staging Metaverse is "Drum Set v<X.X>", where <X.X> is some version number that'll change over time. Click "Purchase".
    i. Verify that you see the "Review Purchase" screen.
    ii. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "CONFIRM PURCHASE" button, and a borderless "CANCEL" button.
2. Click "CONFIRM PURCHASE".
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **entity** [marketplace item] has been added..."
    iii. Verify that you see a "[magic wand icon] REZ" button, and that it is disabled.
    iv. Verify that you DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
3. Click "View this item in My Purchases"
    i. Verify that you are taken to the My Purchases screen, and that the item you just purchased appears at the top of the list.
    ii. Verify that you see a "[magic wand icon] REZ" button on the card of the item you just purchased, and that the button is enabled.
4. Click the item name of the marketplace item you just purchased.
    i. Verify that you are returned to the marketplace item page for the item from (1).
5. Click "Purchase" on the marketplace item page.
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "VIEW THIS ITEM IN MY PURCHASES" button, a grey "BUY IT AGAIN" button, and a borderless "CANCEL" button.
6. Click "VIEW THIS ITEM IN MY PURCHASES"
    i. Verify that you are taken to the My Purchases screen, and that the filter bar contains the text corresponding to the title of the item you were about to buy.
7. Click the "x" icon inside the filter bar.
    i. Verify that the text inside the filter bar disappears.
8. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 1 item appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 0 items appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 0 items appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 0 items appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 0 items appear in the list after you have selected the "Updatable" category in the filter bar.
9. Rez the regular entity.
    i. The entity appears in front of you.
10. Inspect the regular entity.
    i. The wearable's certificate is valid.
11. From the Marketplace, purchase a second copy of the same regular entity as in (1).

### Test C: Avatar Purchase - Base Version
1. Open the Marketplace. Go to the item page for an avatar. One example of a regular entity on the Staging Metaverse is "Finnigon v<X.X>", where <X.X> is some version number that'll change over time. Click "Purchase".
    i. Verify that you see the "Review Purchase" screen.
    ii. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "CONFIRM PURCHASE" button, and a borderless "CANCEL" button.
2. Click "CONFIRM PURCHASE".
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **avatar** [marketplace item] has been added..."
    iii. Verify that you see a "[avatar icon] WEAR" button, and that it is enabled.
    iv. Verify that DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
3. Click the "WEAR" button.
    i. Verify that you see a "Change Avatar" popup with explanitory text. Click "Cancel".
4. Click "View this item in My Purchases"
    i. Verify that you are taken to the My Purchases screen, and that the item you just purchased appears at the top of the list.
    ii. Verify that you see a "[avatar icon] WEAR" button on the card of the item you just purchased, and that the button is enabled.
5. Click the item name of the marketplace item you just purchased.
    i. Verify that you are returned to the marketplace item page for the item from (1).
6. Click "Purchase" on the marketplace item page.
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "VIEW THIS ITEM IN MY PURCHASES" button, and a borderless "CANCEL" button.
7. Click "VIEW THIS ITEM IN MY PURCHASES"
    i. Verify that you are taken to the My Purchases screen, and that the filter bar contains the text corresponding to the title of the item you were about to buy.
8. Click the "x" icon inside the filter bar.
    i. Verify that the text inside the filter bar disappears.
9. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 2 items appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 1 item appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 0 items appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 0 items appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 0 items appear in the list after you have selected the "Updatable" category in the filter bar.
10. Click the "WEAR" button on the card corresponding to the avatar you just purchased.
    i. Verify that you see a "Change Avatar" popup with explanitory text.
11. Click "CONFIRM".
    i. Verify that your avatar changed (by, for example, zooming out to third person view).
    ii. Verify that the "WEAR" button you just clicked changed to be a "[avatar icon] CURRENT" button, and that it is disabled.

### Test D: Content Set/Environment Purchase - Base Version
1. Open the Marketplace. Go to the item page for a content set. One example of a regular entity on the Staging Metaverse is "Oasis v<X.X>", where <X.X> is some version number that'll change over time. Click "Purchase".
    i. Verify that you see the "Review Purchase" screen.
    ii. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "CONFIRM PURCHASE" button, and a borderless "CANCEL" button.
2. Click "CONFIRM PURCHASE".
    i. Verify that you see a warning popup that explains that you need 'replace content' permission.
3. Click "CONFIRM".
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **content set** [marketplace item] has been added..."
    iii. Verify that you see a "[globe icon] REPLACE CONTENT SET" button, and that it is enabled.
    iv. Verify that DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
4. Click "View this item in My Purchases"
    i. Verify that you are taken to the My Purchases screen, and that the item you just purchased appears at the top of the list.
    ii. Verify that you see a "[globe icon] REPLACE" button on the card of the item you just purchased, and that the button is enabled.
5. Click "Purchase" on the marketplace item page.
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "VIEW THIS ITEM IN MY PURCHASES" button, a grey "BUY IT AGAIN" button, and a borderless "CANCEL" button.
6. Click "VIEW THIS ITEM IN MY PURCHASES"
    i. Verify that you are taken to the My Purchases screen, and that the filter bar contains the text corresponding to the title of the item you were about to buy.
7. Click the "x" icon inside the filter bar.
    i. Verify that the text inside the filter bar disappears.
8. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 2 items appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 1 item appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 1 item appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 0 items appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 0 items appear in the list after you have selected the "Updatable" category in the filter bar.
9. Click the "REPLACE" button next to the content set you just purchased.
    i. Verify that you see a "Replace Content" popup. Verify that the information in the popup makes sense.
10. Click the "CONFIRM" button.
    i. Verify that the "REPLACE" button is replaced with a green "REPLACED" overlay for 2 seconds.
    ii. Verify that your Sandbox content has been replaced with the content from the set you just purchased.
11. Right click on any entity that you see in the domain.
    i. Verify that you see an (i) overlay.
12. Click the (i) overlay.
    i. Verify that you see a certificate of the content set you just purchased.
    
### Test E: App Purchase - Base Version
1. Open the Marketplace. Go to the item page for an app. One example of a regular entity on the Staging Metaverse is "App v<X.X> - BLUE VERSION", where <X.X> is some version number that'll change over time. Click "Purchase".
    i. Verify that you see the "Review Purchase" screen.
    ii. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "CONFIRM PURCHASE" button, and a borderless "CANCEL" button.
2. Click "CONFIRM PURCHASE".
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **app** [marketplace item] has been added..."
    iii. Verify that you see a "[download icon] INSTALL" button, and that it is enabled.
    iv. Verify that DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
3. Open your Running Scripts window (CTRL+J) and move it to one side of the Checkout window.
4. On the "Thank you for your purchase" screen, click the "INSTALL" button.
    i. Verify that your "Running Scripts" window shows the script corresponding to the app you just installed.
        - If you have just installed the "BLUE" app, that script should be `simpleApp_blue.js`
    ii. Verify that the "INSTALL" button has changed to "OPEN"
    iii. Close the "Running Scripts" window now.
5. Click the "OPEN" button.
    i. Verify that the app's main screen has opened. Close the app.
6. Open My Purchases.
    i. Verify that the app you just purchased appears at the top of the Purchases list.
    ii. Verify that there are two buttons on that app's card: "OPEN" and "UNINSTALL"
7. Open the Marketplace. Go to the item page for the same app as in (1). Click "Purchase".
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, the price of the item, a blue "VIEW THIS ITEM IN MY PURCHASES" button, and a borderless "CANCEL" button.
8. Click "VIEW THIS ITEM IN MY PURCHASES"
    i. Verify that you are taken to the My Purchases screen, and that the filter bar contains the text corresponding to the title of the item you were about to buy.
    ii. Verify that there are two buttons on that card - "OPEN" and "UNINSTALL"
9. Click the "x" icon inside the filter bar.
    i. Verify that the text inside the filter bar disappears.
10. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 2 items appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 1 item appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 1 item appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 1 item appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 0 items appear in the list after you have selected the "Updatable" category in the filter bar.

### STOP RIGHT THERE FOR A MINUTE
- At this point, talk to Zach. Mention that you're going through the item updates test plan and that you need the items on Staging to be updated. Also send a screenshot of My Purchases (or just tell Zach which items you've purchased from Staging; whichever is easier).
- After you've done this, close your Wallet and the Market app, then twiddle your thumbs for a bit. At this point, Zach will be updating the items you've purchased to newer versions, after which you'll be able to continue with the test plan.
    
### Test F: Filter Bar & "Updates Available" Notification Light Tests
- Ensure that you've performed the above steps (under "STOP RIGHT THERE FOR A MINUTE") before continuing.
- At this point, Zach has uploaded updates to all five Marketplace items that you have purchased.
- Perform this test in both Desktop and HMD modes.

1. Look at your Toolbar.
    i. Verify that your MARKET icon is inactive (black) and that it _DOES NOT_ have a red dot above the cart.
2. Open the WALLET app.
    i. Verify that the MARKET icon is inactive (black) and that it _DOES_ have a red dot above the cart.
    ii. Verify that the "My Purchases" link to the right of "Recent Activity" _DOES NOT_ have a red dot to its left.
3. Open the MARKET app.
    i. Verify that the MARKET icon is active (white) and that it _DOES_ have a red dot above the cart.
    ii. Verify that the "My Purchases" link at the top of the phage _DOES_ have a red dot to its left.
4. Click on the "My Purchases" link at the top of the page. Look at your list of Purchases.
    i. Verify that all five items you purchased in Tests A, B, C, D, and E have banners beneath their cards that mention that there is an update available for that item.
5. Click on the magnifying glass inside the filter bar. Select each category one by one.
    i. Verify that you see 1 item appear in the list after you have selected the "Wearable" category in the filter bar.
    ii. Verify that you see 2 items appear in the list after you have selected the "Entity" category in the filter bar.
    iii. Verify that you see 1 item appear in the list after you have selected the "Avatar" category in the filter bar.
    iv. Verify that you see 1 item appear in the list after you have selected the "Content Set" category in the filter bar.
    v. Verify that you see 1 item appear in the list after you have selected the "App" category in the filter bar.
    vi. Verify that you see 6 items appear in the list after you have selected the "Updatable" category in the filter bar.
6. Click on the magnifying glass inside the filter bar. Select the "Updatable" category.
    i. Type some text into the filter bar and verify that the text you type is correctly filtering the items that appear in My Purchases.
7. While there is still text in the filter bar, force focus loss on the filter bar. (In Desktop mode, click somewhere else. In HMD mode, lower the keyboard.)
    i. Verify that the color of the text inside the filter bar has changed.
8. After typing some text into the filter bar, press the Backspace button until there is no more text in the filter bar, and only the "Updatable" primary filter "pill" remains. Press backspace one more time.
    i. Verify that the "Updatable" primary filter "pill" has disappeared.

### Test G: Item Update Tests
- At this point, Zach has uploaded updates to all five Marketplace items that you have purchased.

1. Open the Marketplace.
    i. Verify that you can no longer find any of the "base" versions of the Marketplace items that you purchased in Tests A, B, C, D, and E.
2. Browse to the Marketplace page of one of the Updated versions of the Marketplace item you purchased. For example, if you purchased "Monztur Hat v2.0", look for "Monztur Hat v3.0". Click "Purchase".
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, "FREE UPDATE", a blue "CONFIRM UPDATE" button, some text about updating, and a borderless "CANCEL" button.
3. Click "Cancel". Click the "My Purchases" link at the top of the page, first verifying that there exists a red dot to its left.
    i. Verify that all items in your Purchases have banners beneath their cards that mention that there is an update available for that item.
4. Click the "Update" link beneath the Wearable Entity that you purchased in Test A.
    i. Verify that you see an image [KNOWN BUG - image won't appear on Staging, do not report!], the title of the item, "FREE UPDATE", a blue "CONFIRM UPDATE" button, some text about updating, and a borderless "CANCEL" button.
5. Click the "CONFIRM UPDATE" button.
    i. Verify that you see the "Thank you for your order!" screen.
    ii. Verify that you see the text "The **wearable** [marketplace item] has been added..."
    iii. Verify that you see a "[hat icon] WEAR" button, and that it is enabled.
    iv. Verify that DO NOT see red text that reads "You do not have Certified Rez permissions in this domain. What does 'Rez' mean?"
6. Close the Market app, then move the first wearable you purchased (in Test A) off to the side a bit. Via My Purchases, Wear the updated version of the wearable. Inspect both wearables.
    i. Verify that the Base version of the wearable that you purchased in Test A now shows an invalid certificate with correct information.
    ii. Verify that the Updated version of the wearable that you just updated to shows a valid certificate with correct information.
7. Open My Purchases.
    i. Verify that now you only see 5 entries in Purchases whose cards have banners beneath them that mention that there is an update available for that item. 
8. Update your Content Set and Avatar.
    i. Verify that everything "works as expected".
9. Go to update your App. Right before you click the "CONFIRM UPDATE" button, open the "Running Scripts" window (CTRL + J) and make sure that the App from Test E is still present. Click "CONFIRM UPDATE".
    i. Verify that the app from Test E has disappeared from your Running Scripts.
10. At this point, you should have only two remaining items in My Purchases. They should both be the same Entity with different Edition Numbers. Choose one Entity to update and remember which Edition it is. Go through the Update process, then return to My Purchases.
    i. Verify that the Entity with the Edition Number that you updated has been replaced with its Updated version. Inspect the updated Entity's certificate and verify that the new Edition Number is the same as the Base Entity's Edition Number.
    ii. Verify that you still have one entity in your Purchases that still has the "Update" banner.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8075/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8075/HighFidelity-Beta-PR12638-eb2f8aa169226c97c6b0e2260896c3e9ffa9376e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8075/HighFidelity-Beta-PR12638-eb2f8aa169226c97c6b0e2260896c3e9ffa9376e.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8097/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8097/HighFidelity-Beta-PR12638-ad33e775f13e475d2405a089f25f6e0cc6fa3d4a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8105/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8106/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8106/HighFidelity-Beta-PR12638-a792bcc3d325609c8f41f0b621a0c06669494472.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8106/HighFidelity-Beta-PR12638-a792bcc3d325609c8f41f0b621a0c06669494472.exe

File: interface/resources/qml/hifi/commerce/checkout/Checkout.qml [hunk 1]
```diff
@@ -49,6 +52,8 @@ Rectangle {
     property bool canRezCertifiedItems: Entities.canRezCertified() || Entities.canRezTmpCertified();
     property string referrer;
     property bool isInstalled;
+    property bool isUpdating;
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/commerce/checkout/Checkout.qml]: We might want to add some comments defining what the various booleans are. Some are about the state of the user/metaverse, some about the state of qml/metaverse operations, some about items, etc., and it isn't always immediately clear from the name.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8136/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8136/HighFidelity-Beta-PR12638-daa68768b957dbfda124458222b312b4738c663a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8136/HighFidelity-Beta-PR12638-daa68768b957dbfda124458222b312b4738c663a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8138/app_12638_PR.apk

**Comment by zfox23** [comment]
Updating test plan and making a small change - please hold...

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8138/HighFidelity-Beta-PR12638-18aa95e0bbb411a9bee6071d61cde24864354a02.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8138/HighFidelity-Beta-PR12638-18aa95e0bbb411a9bee6071d61cde24864354a02.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8145/app_12638_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8145/HighFidelity-Beta-PR12638-0b810c3aa8984d2081c81cfd428ca37a4ce0793d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8145/HighFidelity-Beta-PR12638-0b810c3aa8984d2081c81cfd428ca37a4ce0793d.exe

**Comment by L370** [comment]
@conklin94122 Completed validation of 12638 per the test case listed in this ticket.  All tests passed.  Validation completed on Windows 10 system.  Per @zfox23 the logic exists on the BE, so validating on a single platform will suffice.

**Comment by zfox23** [comment]
DO NOT MERGE until backend component is merged.

**Pull request merged**